packages/server/src/jsRunner/bundles/snippets.ts

```diff
@@ -0,0 +1,19 @@
+// @ts-ignore
+// eslint-disable-next-line local-rules/no-budibase-imports
+import { iifeWrapper } from "@budibase/string-templates/iife"
+
+export default new Proxy(
+  {},
+  {
+    get: function (_, name) {
+      // Snippet definitions are injected to the isolate global scope before
+      // this bundle is loaded, so we can access it from there.
+      // https://esbuild.github.io/content-types/#direct-eval for info on why
+      // eval is being called this way.
+      // @ts-ignore
+      // eslint-disable-next-line no-undef
+      const snippet = (snippetDefinitions || []).find(x => x.name === name)
+      return [eval][0](iifeWrapper(snippet.code))
```

Comment:
As I understand it, this means we're going to `eval` the snippet source every time we call it? Are there negative performance implications of that? What would be wrong with properly compiling the JavaScript when we `compileScriptSync` in `withSnippets`?

Reply:
Absolutely correct, I don't see a reason that we can't optimise this in advance by both your previous suggestion of gathering into an object, and also updating the compiled script source to do a single eval of each snippet. The benefit for a single evaluation is negligible since you're unlikely to use the same snippet more than once, but since we cache the isolate there are definitely performance gains here for things like formulas.

Reply:
Ah actually there is a reason - doing it this way will only eval the snippets that are actually used. If we eval them all up front, you're evaluating every single snippet even if you use none of them. I'm not sure if that is worth updating or not then, since you're potentially degrading performance if you have many (or very large) snippets. I could try a best of both worlds approach, where we cache snippets which have already been eval'd, but only eval them when required like this?

Reply:
I've updated both isolated-vm and the browser to still do evaluation at runtime, but then cache the evaluated snippets for future use. This is the best of both worlds approach that I mentioned above - we only eval snippets that are used, and only once.